security/csrf.rst

```diff
@@ -302,6 +310,160 @@ targeted parts of the plaintext. To mitigate these attacks, and prevent an
+(the default way to declare *your* services) - will make your form types use the
+``submit`` token identifier by default:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/csrf.yaml
+        framework:
+            form:
+                csrf_protection:
+                    token_id: 'submit'
+
+    .. code-block:: xml
+
+        <!-- config/packages/csrf.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony
+                https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:form>
+                    <framework:csrf-protection token-id="submit"/>
+                </framework:form>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/csrf.php
+        use Symfony\Config\FrameworkConfig;
+
+        return static function (FrameworkConfig $framework): void {
+            $framework->form()
+                ->csrfProtection()
+                    ->tokenId('submit')
+            ;
+        };
+
+Forms configured with a token identifier listed in the above ``stateless_token_ids``
+option will use the stateless CSRF protection.
+
+In addition to the ``Origin`` and ``Referer`` headers, stateless CSRF protection
+also checks a cookie and a header (named ``csrf-token`` by default, see the
+:ref:`CSRF configuration reference <reference-framework-csrf-protection>`).
+
+These extra checks are part of defense-in-depth strategies provided by the
+stateless CSRF protection. They are optional and they require
+`some JavaScript`_ to be activated. This JavaScript is responsible for generating
+a crypto-safe random token when a form is submitted, then putting the token in
+the hidden CSRF field of the form and submitting it also as a cookie and header.
+On the server-side, the CSRF token is validated by checking the cookie and header
+values. This "double-submit" protection relies on the same-origin policy
+implemented by browsers and is strengthened by regenerating the token at every
+form submission - which prevents cookie fixation issues - and by using
+``samesite=strict`` and ``__Host-`` cookies, which make them domain-bound and
+HTTPS-only.
+
+Note that the default snippet of JavaScript provided by Symfony requires that
+the hidden CSRF form field is either named ``_csrf_token``, or that it has the
+``data-controller="csrf-protection"`` attribute. You can of course take
+inspiration from this snippet to write your own, provided you follow the same
+protocol.
```

Comment:
Should this document the usage of the `field_attr` option to show how to add such attribute when using the form component ?

Reply:
it's already the default so I guess it's not needed

Reply:
If they write custom JS code (for instance because their project does not use Stimulus), having the example of how to configure it would still be good IMO.

Btw, I'm 99% sure that the usage of `field_attr` is broken in XML, as it does not properly normalizes the XML children to account for the parsing difference between a single child and multiple children, that we need for map fields.

Reply:
Fix https://github.com/symfony/symfony/pull/60439 for the XML issue (no plural needed)
For the JS, the link to the recipe file is enough IMHO. We could expland and tell what it does but I'd wait for someone to feel the need before. Reading the JS is equally informative for now.

Reply:
What I mean is that if they write custom JS, they might need to set a *different* attribute to trigger it.

Reply:
Got it. I'm going to skip this suggestion, this looks niche, and people might figure out on their own. Let's see later if not :)